### ISSUE TITLE
[Bug] NullReferenceException from MSAL when using region with CCA

### ISSUE THREAD

**Issue by trwalke** [AzureAD/microsoft-authentication-library-for-dotnet#2837]
**Logs and network traces**
CorrelationId == "a95592bb-f6c4-4f96-8e09-1ed652ec76fd"

**Which version of MSAL.NET are you using?**
4.35.1.0

**Platform**

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Authentication
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web app 
    * [ ] Authorization code
    * [ ] On-Behalf-Of
* Daemon app 
    * [x] Service to Service calls

Other?

**Is this a new or existing app?**
Upgrade to MSAL

**Repro**
Create CCA
```csharp
//Create CCA
var confidentialClientApplication = ConfidentialClientApplicationBuilder.Create(ClientId)
    .WithCertificate(Cert)
    .WithAzureRegion(msalRegion)
    .WithLegacyCacheCompatibility(false)
    .WithLogging(this.LogMSALMessages, Microsoft.Identity.Client.LogLevel.Warning)
    .Build();

//Acquire Token
var authenticationResult = await this.GetConfidentialClientApplication()
    .AcquireTokenForClient(scopes)
    .WithAuthority($"https://login.microsoft.com/{tenantId}")) // this is the root cause of the failure
    .WithSendX5C(sendX5c)
    .ExecuteAsync()
    .ConfigureAwait(false);
```

**Expected behavior**
MSAL should acquire a token

**Actual behavior**
Null Ref

System.NullReferenceException : Object reference not set to an instance of an object.
   at Microsoft.Identity.Client.Region.RegionManager.IsTelemetryRecorded(ApiEvent apiEvent)
   at Microsoft.Identity.Client.Region.RegionManager.RecordTelemetry(ApiEvent apiEvent, String azureRegionConfig, RegionInfo discoveredRegion)
 ...

Root cause
------------

MSAL tries to validate that the authority at the request level is the same as the authority at the app level. Since the hosts differ, it tries to do instance discovery to find out aliases. There is no api event, so this fails.

**Comment by jmprieur** [comment]
What are the repro steps, @trwalke. is it for a specific region?

**Comment by trwalke** [comment]
@jmprieur gathering more info from the customer at the moment

**Comment by jmprieur** [comment]
@trwalke can you please update the information?

**Comment by bgavrilMS** [comment]
This repros consistently if the authority override (the one at the host level) is different than the one at the client level. I updated the repro steps. This is a P1.

**Comment by trwalke** [comment]
OK, understood, Thanks @bgavrilMS

**Comment by trwalke** [comment]
Released in 4.37.0

**Comment by bgavrilMS** [comment]
Closing as it was released in 4.37
